dawn/src/dawn/CodeGen/CodeGen.cpp

```diff
@@ -29,10 +29,12 @@ std::string CodeGen::generateGlobals(stencilInstantiationContext& context,
                                      std::string outer_namespace_, std::string inner_namespace_) {
 
   std::stringstream ss;
-  Namespace outerNamespace(outer_namespace_, ss);
   std::string globals = generateGlobals(context, inner_namespace_);
-  ss << globals;
-  outerNamespace.commit();
+  if(globals != "") {
```

Comment:
I propose std::optional instead of an empty string

Reply:
`globals` is generated code and we want to wrap it into a namespace. The namespace is of course not needed if there is no code - that would be just an empty namespace. So the code expresses quite well what we want to do.